tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -1142,26 +1148,45 @@ public:
    SumHelper(SumHelper &&) = default;
    SumHelper(const SumHelper &) = delete;
    SumHelper(const std::shared_ptr<ResultType> &sumVPtr, const unsigned int nSlots)
-      : fResultSum(sumVPtr), fSums(nSlots, NeutralElement(*sumVPtr, -1))
+      : fResultSum(sumVPtr) , fSums(nSlots, NeutralElement(*sumVPtr, -1)), fCompensations(nSlots, NeutralElement(*sumVPtr, -1))
    {
-   }
 
+   }
    void InitTask(TTreeReader *, unsigned int) {}
-   void Exec(unsigned int slot, ResultType v) { fSums[slot] += v; }
+
+   void Exec(unsigned int slot, ResultType x)
+   {
+      // Kahan Sum
+      ResultType y = x - fCompensations[slot];
+      ResultType t = fSums[slot] + y;
+      fCompensations[slot] = (t - fSums[slot]) - y;
+      fSums[slot] = t;
+   }
 
    template <typename T, std::enable_if_t<IsDataContainer<T>::value, int> = 0>
    void Exec(unsigned int slot, const T &vs)
    {
-      for (auto &&v : vs)
-         fSums[slot] += static_cast<ResultType>(v);
-   }
+      for (auto &&v : vs) {
+         Exec(slot, v);
+      }
+   }  
 
    void Initialize() { /* noop */}
 
    void Finalize()
    {
-      for (auto &m : fSums)
-         *fResultSum += m;
+   ResultType sum(0) ;
+   ResultType compensation(0);
+   ResultType y(0);
+   ResultType t(0);
```

Comment:
all these `0`s should be `NeutralElement(m, 0)` for generality. Then we could discuss whether this generality still makes sense now that we are strongly assuming that what we are summing are numbers or arrays thereof, but that's for a second step, for now let's keep code consistent.